core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2064,7 +2065,7 @@ struct ViewOffset<
         m_stride(
             Padding<TrivialScalarSize>::
                 stride(/* 2 <= rank */
-                       m_dim.N1 *
```

Comment:
If you do this cast instead on `m_dim.N7`, the 'common type' of each ternary operator expression will be `size_type`, and that will bubble up through the whole expression. Doing it up here seems to assume that any potential overflow doesn't happen until the arithmetic reaches the first dimension.

Reply:
Good point.

Reply:
043aadd7ca525aaeadc909a12e11879784a3246b